### ISSUE TITLE
[MEDIUM] Verify Random State Consistency Across All Notebooks

### ISSUE THREAD

**Issue by lucidopus** [PradHolla/CS559-Final-Project#7]
## Reproducibility: Random State Verification Needed

PDF explicitly requires reproducibility (Page 6):
> 'Make sure to use **random.seed()** or **random_state=** whenever needed so the same results are reproduced each time the notebook runs.'

## Current State
- Most models use `RANDOM_STATE = 42` ✓
- Need to verify ALL random operations use it

### 1. Audit All Notebooks

Check every instance of:
- [ ] `train_test_split()` → has `random_state=42`?
- [ ] Clustering algorithms (KMeans, etc.) → has `random_state=42`?
- [ ] All classifiers → has `random_state=42`?
- [ ] `StratifiedKFold()`, `RepeatedStratifiedKFold()` → has `random_state=42`?
- [ ] `cross_val_score()` → has `random_state=42`?
- [ ] Any sampling operations → has `random_state=42`?
- [ ] LightGBM, XGBoost → has `random_state=42`?

### 2. Standardize Convention
- [ ] Define `RANDOM_STATE = 42` at top of each notebook
- [ ] Use constant throughout (not magic number 42)
- [ ] Add comment explaining why

### 3. Test Reproducibility
- [ ] Run each notebook twice
- [ ] Verify results are identical
- [ ] Check TT, TF values match exactly each run

## Files to Check
- [ ] `Group10_TrainingData.ipynb`
- [ ] `[PersonA]_0.ipynb`
- [ ] `[PersonB]_4.ipynb`
- [ ] `[PersonC]_2.ipynb`
- [ ] `[PersonD]_2.ipynb`
- [ ] `Group10_Generalization.ipynb`

## Common Places to Check
```python
# These all need random_state=RANDOM_STATE
KMeans(n_clusters=7, random_state=RANDOM_STATE)
StratifiedKFold(n_splits=5, random_state=RANDOM_STATE)
RepeatedStratifiedKFold(n_splits=3, n_repeats=10, random_state=RANDOM_STATE)
RandomForestClassifier(random_state=RANDOM_STATE)
cross_val_score(..., cv=cv_strategy, ...)
LGBMClassifier(..., random_state=RANDOM_STATE)
XGBClassifier(..., random_state=RANDOM_STATE)
```

## Acceptance Criteria
- [ ] All random operations have `random_state=RANDOM_STATE`
- [ ] Running notebooks multiple times produces identical results
- [ ] No warnings about non-deterministic behavior
- [ ] TT, TF values match on each run

## References
PDF Section 5.2 (Page 6)
